sound/soc/sof/pcm.c

```diff
@@ -753,15 +756,15 @@ int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd, struct snd_pcm_hw_pa
 	}
 
 	/* read rate and channels from topology */
-	switch (dai->dai_config->type) {
+	switch (private->dai_config->type) {
```

Comment:
For me this is still IPC dependent, the `dai->dai_config` and `dai->comp_dai` IPC message struct just moved under `dai->priv`

Is this an approach which can be used for hw_params (`struct sof_ipc_pcm_params` and `struct sof_ipc_stream_params`) as well? The main problem is that the pcm (and topology for the keyword stream) updates these struct in open code and it is also accessed down in platform code.

Reply:
@ujfalusi I only made the struct IPC agnostic. Its users will need to adapt to make themselves IPC agnostic and I suppose your changes will have to cover that.